vulture/core.py

```diff
@@ -384,6 +384,14 @@ def visit_FunctionDef(self, node):
             if param and isinstance(param, str):
                 self._define_variable(param, node.lineno, confidence=100)
 
+    def visit_If(self, node):
+        if utils.condition_is_unsatisfiable(node.test):
+            self._define(self.unreachable_code, 'if', node.lineno,
+                         size=lines.count_lines(node) if self.sort_by_size
+                         else 1,
+                         message="unsatisfiable 'if' condition",
+                         confidence=100)
+
```

Comment:
What's important here are the tests: 

```
if False:
    unreachable
else:
     reachable
```

```
if foo:
    reachable
elif False:
    unreachable
else:
     reachable
```
-> If a test is unsatisfiable, we report it.

```
if True:
    reachable
else:
     unreachable
```

```
if True:
    reachable
elif foo:
    unreachable
else:
     unreachable
```

```
if foo:
    reachable
elif True:
    reachable
else:
     unreachable
```

-> If a test is always satisfied, all cases after it are dead code.